### ISSUE TITLE
False positive with psuedo-element selectors

### ISSUE THREAD

**Issue by Reinachan** [csstools/stylelint-prefer-nesting#25]
Becasuse of a limitation with CSS nesting, selecting a psuedo-class within a psuedo-element does not work. Basically, this doesn't work

```css
input[type="file"] {
  cursor: pointer;

  &::file-selector-button {
    background-color: grey;

    &:hover {
      background-color: black;
    }
  }
}
```

However, if we try to do this

```css
input[type="file"] {
  cursor: pointer;

  &::file-selector-button {
    background-color: grey;

  }

  &::file-selector-button:hover {
    background-color: black;
  }
}
```

Now I'm getting an error because it's requiring me to use nesting. I'd love to do so, but it's not actually possible.

I tried fixing it using something like this in the config

```js
"csstools/use-nesting": ["always", { except: [/::.*/i] }],
```

However, that doesn't work either. I'm not really sure why, but I can't get the except to work at all. Idk if that's a bug or not, but it's not really relevant to me otherwise, and I think this issue shouldn't require a workaround.

## Solution

I have to possible solutions. The first one is ideal imo and the other is fine.

1. use-nesting should detect when you're trying to use a pseudo-class inside a psuedo-element and move force it to not be nested. (this works great for css files, but is not ideal for scss)

e.g. this

```css
input[type="file"] {
  cursor: pointer;

  &::file-selector-button {
    background-color: grey;

    &:hover {
      background-color: black;
    }
  }
}
```

Should correct to this

```css
input[type="file"] {
  cursor: pointer;

  &::file-selector-button {
    background-color: grey;

  }

  &::file-selector-button:hover {
    background-color: black;
  }
}
```

2. use-nesting should not complain about psuedo-elements with a psuedo-class not being nested (works with any filetype, but doesn't warn about incorrect behaviour in css files)

e.g. use-nesting should not complain about this
```css
input[type="file"] {
  cursor: pointer;

  &::file-selector-button {
    background-color: grey;

  }

  &::file-selector-button:hover {
    background-color: black;
  }
}
```

**Comment by veyndan** [comment]
I've just been having a look into what a fix would look like for this, as I've also been facing this issue and has been causing some headaches.

It may be obvious to everyone else, but I wasn't sure whether i was just using a non-spec compliant browser and this is (theoretically) actually valid CSS, but the lack of support is [called out in the spec](https://drafts.csswg.org/css-nesting-1/#nest-selector), with it being a [known issue they'd like to fix](https://drafts.csswg.org/css-nesting-1/#issue-62c479a8).

Since it's not currently valid CSS, I'll have a go at submitting a fix. Hopefully the issue in the spec can be resolved and the submitted fix can be ripped out!

**Comment by romainmenke** [comment]
This has been fixed in 6.0.2:

https://stylelint.io/demo/#N4Igxg9gJgpiBcICWA7ADgVwC4G0sE80YBeAHRADMkAbGcgXQAJhSVHGwMAnAZwi-iM0EVFhhcA3K1bsAZPHhVaAWh4xaYLP2UAjbFrYs27RjoCGYANYBzLhAwooyyNX6DbMfFJQzGAX2ljeUUaGFV1GE1tPSwDeAALCAA3cWZfdnMrW3tHZwhXAVNqC0tvdgCUPxAAGkpQgDkzAFs4RBgAD2a0WgA6MB4eGvAIFCprBBAO4S4sRlgKMwxqWaN2cm6Ma1QeckEcdMZyHgJaalFlDDVlFBhj1GtyX3pq33IuJdvdtOM18AGtfI8AD0lzCNzuKAeCEOIDM1AA7mZ8DsQIwAD5omFIawofh0EC+Cp+bxDSCjbEAMX4TTMWAmx3wp1EfRGYx6TQAVoNarA0INEKsYQymSgsF9yMUxMdyC9jEcTupzmSxqosGZHGYuFBxSBJbcxTVXiBhYrRRcruCsPcdQA2HoABh6ACZHpVWCA-EA
